server_config_test.cc

```diff
@@ -20,130 +21,243 @@ class ServerConfigTest : public ::testing::Test {
-    std::vector<std::string> query = {"server", "listen"};
+    std::unique_ptr<ServerConfig> server_config = ParseString("server { port; }");
+    bool buildCode = server_config->Build();
+
+    std::vector<std::string> query = {"server", "port"};
     std::string port = "";
-    int ec = server_config->propertyLookUp(query, port);
+    bool ec = server_config->propertyLookUp(query, port);
 
-    EXPECT_EQ(1, ec)
-      << "Error value of 1 should be returned if property isn't found with a value";
+    EXPECT_EQ(false, buildCode)
+      << "Build() should return false when given a property has no value";
+    EXPECT_EQ(false, ec)
+      << "LookUp() should return false if a nested property doesn't have a value";
     EXPECT_EQ("", port)
       << "A listen property without a port number in a nested block should \
           give an empty lookup result";
 }
 
-TEST_F(ServerConfigTest, NonsenseProperty) {
-    std::unique_ptr<ServerConfig> server_config = ParseString("listen 1234;");
+TEST_F(ServerConfigTest, NonsenseQueryPath) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("port 1234;");
+    bool buildCode = server_config->Build();
+
     std::vector<std::string> query = {"laksdj3"};
     std::string port = "";
-    int ec = server_config->propertyLookUp(query, port);
+    bool ec = server_config->propertyLookUp(query, port);
 
-    EXPECT_EQ(1, ec)
-      << "A property lookup on a nonsense property should return 1 (meaning error)";
+    EXPECT_EQ(true, buildCode)
+      << "Build() should return true when given a valid config";
+    EXPECT_EQ(false, ec)
+      << "A property lookup on a nonsense property should return false";
 }
 
 TEST_F(ServerConfigTest, SingleNestedPort) {
     std::unique_ptr<ServerConfig> server_config
-      = ParseString("server { listen 1234; }");
-    std::vector<std::string> query = {"server", "listen"};
+      = ParseString("server { port 1234; }");
+    bool buildCode = server_config->Build();
+
+    std::vector<std::string> query = {"server", "port"};
     std::string port = "";
-    int ec = server_config->propertyLookUp(query, port);
+    bool ec = server_config->propertyLookUp(query, port);
 
-    EXPECT_EQ(0, ec)
-      << "Error value of 0 should be returned if property is found";
+    EXPECT_EQ(true, buildCode)
+      << "Build() should return true when given a valid config";
+    EXPECT_EQ(true, ec)
+      << "LookUp() should return true if property is found";
     EXPECT_EQ("1234", port)
       << "A listen property with a port number in a nested block should \
           yield a port number";
 }
 
 TEST_F(ServerConfigTest, DoublyNestedPort) {
     std::unique_ptr<ServerConfig> server_config
-      = ParseString("outer { server { listen 1234; } }");
-    std::vector<std::string> query = {"outer", "server", "listen"};
+      = ParseString("outer { server { port 1234; } }");
+    bool buildCode = server_config->Build();
+
+    std::vector<std::string> query = {"outer", "server", "port"};
```

Comment:
Could we add a test case for where the query-string is missing the `outer` (and should fail)? That would let us guard against the regression where we inadvertently introduce an assumed outer scope, for example.